Removes the star icon font

erikyo opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This Pr will replace the WooCommerce font with a new one that is the merge of the previous WooCommerce Font and the Star font. In this way we need to load only a single font to get the same result. 

Closes #31490 

### How to test the changes in this Pull Request:

1. Rebuild assets (this is mandatory because i've changed _font.scss and the other theme related )
2. To know if the patch is working delete all the Star.* fonts from /woocommerce/assets/fonts/star.woff, if the star icon doesn't disappear the font family was correcty replaced. rel to [_fonts.scss](https://github.com/erikyo/woocommerce/blob/remove-star-icon/plugins/woocommerce/legacy/css/_fonts.scss)
3. Check in DevTools under the network panel, filtering fonts. Star.* should not be present
4. While testing .svg font remember that are allowed only before chrome 38, and actually works only in Safari. [source](https://caniuse.com/svg-fonts). The reason was a poor rendering quality, and in addition, some features are missing eg. ligatures. Same for the eot font that can be tested only with IE <= 11


### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [x] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Removes star font

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>


#### Testing Instructions

1. Begin by setting up a fresh WP/WooCommerce installation, followed by modifying the theme you have selected.
2. Access the font files from [this location](https://github.com/woocommerce/woocommerce/tree/ce03a12790c526e88e382d62dc170b63db7a0884/plugins/woocommerce/assets/fonts) and store them within a folder titled "fonts." Integrate this folder into the root directory of your active theme.
3. Integrate the following code into your `function.php` file:
```php
function add_a_font() {
?>
  <style>
  @font-face {
      font-family: 'new-star';
      src: url('fonts/WooCommerce.woff2') format('woff2'),
      url('fonts/WooCommerce.woff') format('woff'),
      url('fonts/WooCommerce.ttf') format('truetype');
      font-weight: normal;
      font-style: normal;
  }
  </style>
<?php 
}
add_action( 'wp_head', 'add_a_font' );
```
4. Proceed to your website and, using developer tools, inspect a WooCommerce page where the star font is displayed. This may vary depending on your chosen theme, but typically, many themes showcase this font within the review tab of the product page. Once you've located the "star" on your page, toggle the "new-star" font using the developer tools. Verify that there are no discernible changes when adding or removing the "new-star" font. (For thoroughness, please ensure your browser is zoomed in to at least 200% to confirm consistency in shape, size, etc.)

thanks!

erikyo committed:
removes the star font
The WooCommerce font can hold many other icons there isn't the reason to load a font for a single icon

erikyo committed:
updates custom templates stylesheets

erikyo committed:
adds woff2 font format to themes customized stylesheets

erikyo committed:
fix the star font size diff

peterfabian reviewed (dismissed):
Hey, thanks for the updated PR! 

Do you know about any eot file viewers I could use to check the changes? It seems a bit odd to me that star.eot is just 1.6KB, but WooCommerce.eot increased by 26KB.

Taking a step back here a bit, since WP no longer supports IE, based on [this post](https://css-tricks.com/snippets/css/using-font-face-in-css/#aa-deepest-possible-browser-support), it seems we don't need to link to eot file anymore. The eot file is fairly large and since there is no `format` hint, the font file will always get downloaded (based on [this docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/src)).

I'd say we can go with the "[Slightly deeper browser support](https://css-tricks.com/snippets/css/using-font-face-in-css/#aa-slightly-deeper-browser-support)" here and then we can also remove the SVG and embedded-opentype:

```
@font-face {
  font-family: 'MyWebFont';
  src: url('myfont.woff2') format('woff2'),
       url('myfont.woff') format('woff'),
       url('myfont.ttf') format('truetype');
}
```

This also means we probably no longer need to change the WooCommerce.eot and SVG file, so only ttf, woff and woff2 needs an update.

Even Twenty Twenty theme from 2 years ago [only includes woff2 format](https://themes.trac.wordpress.org/browser/twentytwenty/1.9/style.css#L261), so it should be safe for us to reduce the coverage. What do you think?

erikyo commented:
> Do you know about any eot file viewers I could use to check the changes? It seems a bit odd to me that star.eot is just 1.6KB, but WooCommerce.eot increased by 26KB.

Sincerely I haven't find no one that work 😅... I've tested the font on a real page with windows 7 / internet explorer to get sure it works but I didn't realize it had become so big. The reason could be some settings used during the conversion with the linux package called "eot-utils" and, maybe, by changing something I can "compress" it more. 

> This also means we probably no longer need to change the WooCommerce.eot and SVG file, so only ttf, woff and woff2 needs an update.

You are right! The eot font is completely useless in 2022 and could be safely removed. Nowdays woff2 is the preferred by most of the browser and woff covers all browsers above ie9. ttf I don't think it will ever be used but maybe it can be useful for other purposes.
https://caniuse.com/mdn-css_at-rules_font-face_woff

One last thing before i'll update the PR... isn't worth to import the [_font.scss file (like here)](https://github.com/erikyo/woocommerce/blob/feature/css-custom-props-r/plugins/woocommerce/legacy/css/twenty-nineteen.scss#L26) into all the Twenty-* style? Currently the code @font-face is repeated for each style (the only difference I noticed is the different kind of quotes) but can be imported (and this avoids in the future having to edit all the files as I'm doing now).

erikyo committed:
removes WooCommerce.eot and SVG file that are no longer needed and updates the template style in order to import from font.scss

samueljseay commented:
@peterfabian friendly ping, how should we move forward on this one.

peterfabian commented:
Thanks for the ping. I'll try to have a look at this within a week or so, but this still feels like a fairly low prio enhancement, so it can take a bit of time until I get back to this.

lsinger reviewed (changes requested):
> One last thing before i'll update the PR... isn't worth to import the [_font.scss file (like here)](https://github.com/erikyo/woocommerce/blob/feature/css-custom-props-r/plugins/woocommerce/legacy/css/twenty-nineteen.scss#L26) into all the Twenty-* style? Currently the code @font-face is repeated for each style (the only difference I noticed is the different kind of quotes) but can be imported (and this avoids in the future having to edit all the files as I'm doing now).

I'd say that would make sense -- several already do that. 

If you're still available to work on this PR, please make the discussed changes and feel free to make the imports consistent across the themes. 

Again, sorry that we've been dropping the ball here. Please let us know if you can still make these changes.

erikyo committed:
Optimized font import
Unifies the import of fonts by extending it to the twenty-twenty-three theme

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

lsinger commented on plugins/woocommerce/assets/fonts/WooCommerce.svg:
```diff
@@ -1 +1,256 @@
-<svg xmlns="http://www.w3.org/2000/svg"><defs><font horiz-adv-x="1024"><font-face ascent="960" descent="-64" units-per-em="1024"/><glyph horiz-adv-x="512"/><glyph d="M933.926 716.642 512 960 319.21 849.383l429.827-233.877zM709.531 593.383 279.704 825.679 88.494 715.062 512 485.926zM512 441.679 67.951 681.877V192L512-64l444.049 256v489.877L512 441.679zm-91.654-270.222c-11.062-6.321-23.704-4.741-39 ...
\ No newline at end of file
+<?xml version="1.0" standalone="no"?>
```
There's lots of meta data in here that we shouldn't include. Please run this through https://github.com/ImageOptim/ImageOptim -- that would remove a few things and also optimize the size a bit.

erikyo replied:
I'm sincerely not sure if SVGO can optimize safely svg font format
https://github.com/svg/svgo/issues/992#issuecomment-1811125790

how about if I manually remove unnecessary meta like: 

```xml
<?xml version="1.0" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
```

and 

```xml
<metadata>
Created by FontForge 20201107 at Fri Feb 18 01:23:36 2022
 By Dell7010

</metadata>
```

the rest of definitions seems fine to me (but i may be wrong, tell me something in this case)

erikyo replied:
In order to fix I proceeded in this way:
- I took the old font svg and added the star glyphs to it... 
- Initially I copied them from the original font (star.svg) but then I realized that the two fonts have different units / ratio
- so I copied the paths (the "d") from the svg font I had generated which was adapted to the bounding box of the woocommerce font. 
- The original star font has no defined names for the "glyph-name" attribute so i added "star" and "star-outline" as the name of the glyph. Tell me if you prefer a different wording for those icons

The result looks correct to me, you can checkout using [fontello](https://fontello.com/)

lsinger commented on plugins/woocommerce/changelog/31670-remove-star-icon:
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: performance
+
+CYS - Fix activeThemeHasMods undefined error.
```
The contents of this changelog file seem off -- maybe a rebase gone wrong? 🤔

erikyo replied:
I've copy pasted the issue template to fire the action because this issue was too old, and I hadn't updated automatically. hope next commit will be fixed

lsinger reviewed (changes requested):
Thanks @erikyo! I left a few comments on a few lines and also have some overarching notes: 

- Just like previous reviewers I'm not comfortable with the 290% increase in `WooCommerce.eot`. What options do we have here? E.g., is there an optimization that we can run? 
- When checking out the branch locally, all the star.* files are still there, and indeed I can't fine a commit that would delete them. Is this intentional? If so, what's the rationale here? 
- Can you suggest an application that can view `eot` files? I tried FontForge but it doesn't seem to like these. Before approving these changes I'd like to take a look at all the binary files directly. 
- For the review instructions, can you list a few pages / views that I can use to verify the fonts still work as intended?

erikyo commented:
> Just like previous reviewers I'm not comfortable with the 290% increase in `WooCommerce.eot`. What options do we have here? E.g., is there an optimization that we can run?

I've edited the font with fontforge that allows to open and export any type of font [excluded the eof](https://github.com/fontforge/fontforge/issues/1468). For that i converted the font using a linux [cli tool](https://www.w3.org/Tools/eot-utils/)... But even I don't know how to read and validate it, there are tools online but they don't read either yours or mine version. 
Anyhow i want to mention that I don't believe that font is relevant anymore, if the problem is the increased file size I suggest deleting it forever

> When checking out the branch locally, all the star.* files are still there, and indeed I can't fine a commit that would delete them. Is this intentional? If so, what's the rationale here?

I would not like to delete the files that are used by third-party plugin, some plugin could rely on that font and I would break the existing stuff. the important thing is that that font is not loaded in the frontend (for this i am loading the same font for the "woocommerce" and "star" font-families

> Can you suggest an application that can view `eot` files? I tried FontForge but it doesn't seem to like these. Before approving these changes I'd like to take a look at all the binary files directly.

I think in 2024 by now no one will be able to view that kind of font 😅, maybe only ie old versions, but as in the first answer  my suggestion is to remove that font (or if you want, please encode the `.ttf` file into a `.eot` with this https://www.w3.org/Tools/eot-utils/ and update my PR)

> For the review instructions, can you list a few pages / views that I can use to verify the fonts still work as intended?

You can check the icon in the review tab of the single page and below the product photo in the archive. if I remember correctly some widgets also used that font as the "woocommerce/all-reviews"

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

erikyo committed:
Svg font minification (WooCommerce.svg)

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

lsinger reviewed (changes requested):
> Anyhow i want to mention that I don't believe that font is relevant anymore, if the problem is the increased file size I suggest deleting it forever

As [even WordPress itself doesn't support Internet Explorer anymore](https://make.wordpress.org/core/handbook/best-practices/browser-support/) -- the only browser to ever have supported [EOT font files](https://en.wikipedia.org/wiki/Embedded_OpenType) -- I'd say we remove the file from WooCommerce. 👍

> I would not like to delete the files that are used by third-party plugin, some plugin could rely on that font and I would break the existing stuff. the important thing is that that font is not loaded in the frontend (for this i am loading the same font for the "woocommerce" and "star" font-families

Great points -- agreed. 👍 I still think we should stop using the "star" font family even if we leave it as defined -- a few `twenty-*.scss` still reference it. See below. 

I tested the changes and everything seems to work well. However, after searching the monorepo for `font-family: star` again, I still found SCSS rules that refer to this font family. If we truly want to phase this specialized font family out we should update those as well, don't you think? 

I'm also adding @gigitux  because there are a few of these references in WooCommerce Blocks and Luigi is listed as the author of at least one of those lines. @gigitux, what do you think -- with the WooCommerce font updated to include the star, can we remove the reference to the "star" font family from WooCommerce Blocks as well?

erikyo commented:
> If we truly want to phase this specialized font family out we should update those as well

Yep, you are right! I can still find 11 occurrences of that font-family, and you also correct about some are from the woocommerce-block repository (eg. plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/rating-stars/style.scss).

If this is what you come up with as well, I will proceed with these substitutions

![image](https://github.com/woocommerce/woocommerce/assets/8550908/ae00760e-2da8-4f99-a13c-71fb61bd1b12)

lsinger commented:
> If this is what you come up with as well

@erikyo yep, that's exactly what I'm seeing as well, even without the `*.scss` filter. 👍

erikyo committed:
"star" font phasing out
The font family 'star' was replaced in all files. However, it continues to remain physically in the repo but not in the style files

erikyo commented:
I did a final check and realised that in some themes (twenty-nineteen.scss,twenty-nineteen.scss,woocommerce.scss) the font-family was enclosed in quotes, so I replaced those as well. 

The only remaining occurrences are in the css (generated, I won't change it), I can't find that font anywhere else now.

gigitux commented:
Thanks, @erikyo, for working on this! 

> I'm also adding @gigitux because there are a few of these references in WooCommerce Blocks and Luigi is listed as the author of at least one of those lines. @gigitux, what do you think -- with the WooCommerce font updated to include the star, can we remove the reference to the "star" font family from WooCommerce Blocks as well?

I can confirm that everything works okay on WooCommerce Blocks:

| Product Filter Block | Single Product Template |
|--------|--------|
| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/b419fda1-87ba-496c-8de3-d7182ae02a39)| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/f8d0ff29-73b0-4831-b7fb-12b63ee318b9) | 


> As [even WordPress itself doesn't support Internet Explorer anymore](https://make.wordpress.org/core/handbook/best-practices/browser-support/) -- the only browser to ever have supported [EOT font files](https://en.wikipedia.org/wiki/Embedded_OpenType) -- I'd say we remove the file from WooCommerce. 👍

I agree with this! Let's remove .oet font given that[ any modern browser supports it](https://caniuse.com/?search=eot) (TIL):

![image](https://github.com/woocommerce/woocommerce/assets/4463174/db63eb95-696b-458e-9eb2-daf03c8f6e23)

erikyo committed:
Removes the `.eot` font file since nowdays it's completely useless

lsinger reviewed (approved):
Took another look at the changes and tried things out. In my testing everything still works as intended. Let's ship this. 🚢
